### PR TITLE
Bar chart domain

### LINKED ISSUE #163: Add 'domain[0, null]' as default for the linear axis

### PR DESCRIPTION
**Pull request by samussiah**
# Issues
Closes #163

### CODE REVIEW

File: src/chart/draw/consolidateData/transformData.js [hunk 1]
```diff
@@ -354,16 +186,33 @@ export default function transformData(raw, mark) {
...
         ? config.y_dom
         : config.y.type === 'ordinal' && config.y.behavior === 'flex'
           ? set(filtered.map(m => m[config.y.column])).values()
           : config.y.type === 'ordinal'
             ? set(raw.map(m => m[config.y.column])).values()
-            : config.y_from0 ? [0, max(pre_y_dom)] : pre_y_dom;
-
+            : pre_y_dom;
+
+    //set lower limit of linear domain to 0 when other axis is ordinal, provided no values are negative
```
Comment (jwildfire) [inline src/chart/draw/consolidateData/transformData.js]: I think this does it. Would probably be good to explain this behavior for "flex bar charts" in the wiki, and note that users can still explicitly set non-zero domains on bar charts by using `behavior` other than flex.

File: src/chart/draw/consolidateData/transformData/makeNest.js [hunk 2]
```diff
@@ -0,0 +1,174 @@
+import { nest, scale, extent, range, ascending, quantile, sum, merge, descending } from 'd3';
```
Comment (jwildfire) [inline src/chart/draw/consolidateData/transformData/makeNest.js]: Assuming this just a copy/paste and not a significant refactor?

File: test/chart/setDomain.js [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+            });
+        });
+
+        //linear x ordinal
+        describe('setDomain() is called for a chart with an linear x-axis and a ordinal y-axis', () => {
+            it('sets the lower limit of the linear domain to 0', () => {
+                expect(charts.linear_ordinal.x_dom[0]).toEqual(0);
+            });
+
+            it('sets the upper limit of the linear domain to the maximum value of the linear axis variable', () => {
```
Comment (jwildfire) [inline test/chart/setDomain.js]: Is it worth adding a test where `behavior` isn't set to `flex` and checking that it is still possible to have a non-zero lower limit for bars?

File: src/chart/draw/consolidateData/transformData/makeNest.js (hunk 2 above)
Comment (samussiah) [reply src/chart/draw/consolidateData/transformData/makeNest.js]: Yeah, I'm afraid to touch it.

File: test/chart/setDomain.js (hunk 3 above)
Comment (samussiah) [reply test/chart/setDomain.js]: So like, verify that without a user-defined domain and with `behavior` unspecified or set to `raw` that the lower limit is 0?

Comment (jwildfire) [reply test/chart/setDomain.js]: I was actually thinking that we want to confirm that *with* a user defined domain (e.g. [10, 20]) and behavior 'flex' the lower limit is *not* 0. (e.g. 10)

Comment (samussiah) [reply test/chart/setDomain.js]: https://github.com/RhoInc/Webcharts/issues/202

**Comment by danedexF5** [comment]
Using the Iris dataset and Master branch, the x-axis starts at 3 and shows 3 continuously at each hash mark. When the branch is switched to bar-chart-domain the x-axis starts at 0.
